x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_signal.ts

```diff
@@ -46,15 +46,19 @@ export const buildSignal = (doc: SignalSourceHit, rule: Partial<RulesSchema>): S
   const ruleWithoutInternalTags = removeInternalTagsFromRule(rule);
   const parent = buildAncestor(doc, rule);
   const ancestors = buildAncestorsSignal(doc, rule);
-  const signal: Signal = {
+  let signal: Signal = {
     parent,
     ancestors,
     original_time: doc._source['@timestamp'],
     status: 'open',
     rule: ruleWithoutInternalTags,
   };
   if (doc._source.event != null) {
-    return { ...signal, original_event: doc._source.event };
+    signal = { ...signal, original_event: doc._source.event };
+  }
+  if (doc._source.threshold_count != null) {
+    signal = { ...signal, threshold_count: doc._source.threshold_count };
+    delete doc._source.threshold_count;
```

Comment:
Why we need to delete that?

Reply:
I believe it's so it doesn't get copied over to the root signal object (below) as it's an additional field unique to the aggregation query and is only intended to be stored within the `signal.threshold_count` field:

https://github.com/elastic/kibana/blob/4974834ed1437b7da1895c03777c9dff01ec4771/x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_bulk_body.ts#L63